frontend/ui/widget/networksetting.lua

```diff
@@ -265,7 +265,8 @@ end
 
 function NetworkItem:saveAndConnectToNetwork(password_input)
     local new_passwd = password_input:getInputText()
-    if new_passwd == nil or string.len(new_passwd) == 0 then
+    -- Dont set a empty password if WPA encryption, go through if it’s an open AP
+    if (new_passwd == nil or string.len(new_passwd) == 0) and string.find(self.info.flags, "WPA") then
```

Comment:
Ah, okay, lots of existing weird syntax in there ;). So, `#` syntactic sugar for length operations, please ;).